README.md

```diff
@@ -8,3 +8,155 @@ development.
+a `Service` configured with an `ExternalName`; you can update this to point to
+the DNS name of your redis service.
+
+For more complex cases, please refer to the following
+[blog post](https://cloud.google.com/blog/products/gcp/kubernetes-best-practices-mapping-external-services)
+on best practices for mapping external services. In general, the only
+requirement of your solution is that you have a service named `redis` in the
+same namespace as your Spinnaker installation that routes to a valid `redis`
+backend.
+
+Regardless of the approach you choose, add all the relevant redis Kubernetes
+objects to your customization via `kustomze edit add resource redis/*.yml`.
+
+#### Add any secret files
+
+The `secrets` folder is intended to hold any secret files that are referenced by
+your config files. The files in this folder will be mounted in `/var/secrets` in
+each pod.
+
+To add a secret, put it in the `secrets/` folder and add it to the `files` field
+of the `spinnaker-secrets` entry in the `kustomization.yaml`. In this example,
+we'll add a _kubeconfig_ file called `k8s-kubeconfig`:
+
+```shell script
+cp <path to kubeconfig> secrets/k8s-kubeconfig
+```
+
+and update the secret in the `kustomization.yaml` file as:
+
+```yaml
+- behavior: merge
+  name: spinnaker-secrets
+  files:
+    - secrets/k8s-kubeconfig
+```
+
+#### Generate Spinnaker config files
+
+This installation pathway expect the config files for your services to be in the
+`kleat` directory. As the name suggests, the expectation is that these files
+will be generated by `kleat` and written to this directory.
+
+`kleat` takes as input a
+[halconfig](https://github.com/spinnaker/kleat/blob/master/docs/docs.md). This
+file is in general compatible with the file consumed by Halyard; if you are
+migrating from Halyard, please read the minor changes you will need to make to
+your file in the [kleat readme](https://github.com/spinnaker/kleat).
+
+Unlike Halyard, `kleat` allows you to specify only the fields that you'd like to
+configure. In this case, we will assume a very simple config:
+
+```yaml
+providers:
+  kubernetes:
+    enabled: true
+    accounts:
+      - name: k8s
+        kubeconfigFile: /var/secrets/k8s-kubeconfig
+    primaryAccount: k8s
+```
+
+Note that the `kubeconfigFile` argument here points to a file relative to
+`/var/secrets`. This is important, as the `kleat` config requires that all file
+references be relative to where they will be mounted in the running container
+(rather than where they happen to be on the machine running kleat).
+
+Save this file as `halconfig.yml` to the current directory, Invoke `kleat` to
+consume the input `halconfig.yml` and output the service configs to the `kleat/`
+directory:
+
+```shell script
+kleat halconfig.yml kleat/
+```
+
+This will generate service configs in the `kleat/` directory, replacing the
+empty configs that were there before. The `kustomization.yaml` file is already
```

Comment:
I noticed that yesterday too...I *think* I did that because all the docs for `kustomize` talk about a `.yaml` file but most of the Kubernetes ones are `.yml`.  But it does seem strange so maybe we should be consistent...should we go with `.yml`?

Reply:
I was about to say yes but then I found: https://yaml.org/faq.html

Reply:
We always use `.yml` in Kleat...

Reply:
Let's do `.yml`, we can always change later if we want.  I'll make a pull request.